src/gui/symbology/qgssvgselectorwidget.cpp

```diff
@@ -396,10 +399,33 @@ QgsSvgSelectorWidget::QgsSvgSelectorWidget( QWidget *parent )
   mGroupsTreeView->setHeaderHidden( true );
   populateList();
 
-  connect( mImagesListView->selectionModel(), &QItemSelectionModel::currentChanged,
-           this, &QgsSvgSelectorWidget::svgSelectionChanged );
-  connect( mGroupsTreeView->selectionModel(), &QItemSelectionModel::currentChanged,
-           this, &QgsSvgSelectorWidget::populateIcons );
+  mParametersModel = new QgsSvgParametersModel( this );
+  mParametersTreeView->setModel( mParametersModel );
+  mParametersGroupBox->setVisible( mAllowParameters );
+
+  mParametersTreeView->setItemDelegateForColumn( static_cast<int>( QgsSvgParametersModel::Column::ExpressionColumn ), new QgsSvgParameterValueDelegate( this ) );
+  mParametersTreeView->header()->setSectionResizeMode( QHeaderView::ResizeToContents );
+  mParametersTreeView->header()->setStretchLastSection( true );
+  mParametersTreeView->setSelectionBehavior( QAbstractItemView::SelectRows );
+  mParametersTreeView->setSelectionMode( QAbstractItemView::SingleSelection );
+  mParametersTreeView->setEditTriggers( QAbstractItemView::DoubleClicked );
+
+  connect( mParametersModel, &QgsSvgParametersModel::parametersChanged, this, &QgsSvgSelectorWidget::svgParametersChanged );
+  connect( mImagesListView->selectionModel(), &QItemSelectionModel::currentChanged, this, &QgsSvgSelectorWidget::svgSelectionChanged );
+  connect( mGroupsTreeView->selectionModel(), &QItemSelectionModel::currentChanged, this, &QgsSvgSelectorWidget::populateIcons );
+  connect( mAddParameterButton, &QToolButton::clicked, mParametersModel, &QgsSvgParametersModel::addParameter );
+  connect( mRemoveParameterButton, &QToolButton::clicked, this, [ = ]()
+  {
+    const QModelIndexList selectedRows = mParametersTreeView->selectionModel()->selectedRows();
+    if ( selectedRows.count() > 0 )
+      mParametersModel->removeParameter( selectedRows.at( 0 ) );
```

Comment:
Might be worth tweaking the code to allow for more than one parameter to be removed per button click.

Reply:
done in 1562a49e0f